src/main/java/io/rpg/model/actions/Battle.java

```diff
@@ -0,0 +1,26 @@
+package io.rpg.model.actions;
+
+import io.rpg.model.object.GameObject;
+import io.rpg.model.object.Player;
+
+public class Battle {
+    private final Player player;
+    private final GameObject opponent;
+    private final int reward;
+
+    public Battle(Player player, GameObject opponent, int reward) {
+        this.player = player;
+        this.opponent = opponent;
+        this.reward = reward;
+    }
+
+    public String action(){
```

Comment:
We could consider returning some kind of `BattleResult` instead of a plain string. Message can be then determined in view layer based on battle result (maybe consider leveraging existing `Result` class). I believe this approach is more flexible and easier to test.